lib/promscrape/targetstatus.qtpl

```diff
@@ -34,15 +34,153 @@ job={%q= jobName %} (0/0 up)
+                                                        {%= formatLabel(promrelabel.FinalizeLabels(nil, ts.sw.Config.Labels)) %}
+                                                    </div>
+                                                    <div style="display:none" id="original_labels_{%s targetID %}">
+                                                        {%= formatLabel(ts.sw.Config.OriginalLabels) %}
+                                                    </div>
+                                                </td>
+                                                <td>{%d ts.scrapesTotal %}</td>
+                                                <td>{%d ts.scrapesFailed %}</td>
+                                                <td>
+                                                    {% if lastScrapeTime < 365*24*time.Hour %}
+                                                    {%f.3 lastScrapeTime.Seconds() %}s ago
+                                                    {% else %}
+                                                    none
+                                                    {% endif %}
+                                                <td>{%d int(ts.scrapeDuration) %}ms</td>
+                                                <td>{%d ts.samplesScraped %}</td>
+                                                <td>{% if ts.err != nil %}{%s ts.err.Error() %}{% endif %}</td>
+                                            </tr>
+                                            {% endfor %}
+                                            </tbody>
+                                        </table>
+                                    </div>
+                                </div>
+                            </div>
+                            {% endfor %}
+                        </div>
+                    </div>
+
+                      {% for _, jobName := range emptyJobs %}
+                        <div>
+                          <h4>
+                            <a>{%s jobName %} (0/0 up)</a>
+                          </h4>
+                          <table class="table table-striped table-hover table-bordered table-sm">
+                            <thead>
+                              <tr>
+                                <th scope="col">Endpoint</th>
+                                <th scope="col">State</th>
+                                <th scope="col">Labels</th>
+                                <th scope="col">Last Scrape</th>
+                                <th scope="col">Scrape Duration</th>
+                                <th scope="col">Samples Scraped</th>
+                                <th scope="col">Error</th>
+                              </tr>
+                            </thead>
+                          </table>
+                        </div>
+                      {% endfor %}
+                </main>
+            </div>
+        </div>
+
+</body>
+<script>
 function collapse_all() {
```

Comment:
It is better to define `collapse_all()` function in front of `<body>`, since otherwise `collapse` functionality won't work until the full `<body>` is loaded. This may be especially visible when `/targets` page contains thousands of targets and it is loaded over low-bandwidth network.

Reply:
in that case in scripts we are trying to find elements which not in DOM at that moment.